### ISSUE TITLE
Why do we avoid busy wait for JDT Indexer in JdtTypeProvider.findSecondaryType

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-eclipse#223]
In JdtTypeProvider.findSecondaryType we do not ask the jdt index while the indexer is still running to avoid waiting for the indexer. Instead we do query the packages and the compilation units manually which might become really slow depending on the situation. Since we do not query the index the indexer will be busy doing nothing (see https://bugs.eclipse.org/bugs/show_bug.cgi?id=514653) and will not finish in a reasonable time span. 

This leads in your situation to a build time that is > 25 mins (instead of 7 without the busy wait avoidance)

If the bug in jdt does not get fixed we could at least do the busy wait in non ui thread case and/or at least introduce a jvm option that let the user change/influence the behaviour

**Comment by svenefftinge** [comment]
I haven't followed up on it, but what is the state with the new JDT index that google was working on?
Maybe that will be a more sustainable solution here?

**Comment by cdietrich** [comment]
it will by contained in 4.7 but will not be the default. am i right @iloveeclipse

**Comment by iloveeclipse** [comment]
Yep. AFAIK the new index will be used *additionally* to the old one, at least this is the current state in M7 and it will be surprising if they would switch the old one off in RC candidate.
